### PR TITLE
Export evaporative flux from ATM

### LINKED ISSUE #671: Add Export of evaporative flux for use by ocean
## Description

MOM6 requires the evaporative flux from the ATM. Currently, the latent heat flux from ATM is converted in CMEPS to provide the required field. It would be better if ATM would provide the evaporative flux directly. 

See https://github.com/NOAA-EMC/CMEPS/issues/93

## Solution

In ``atmos_model``, add a conversion of the latent heat flux (``dqsfc_cpl``) to provide a new output for the evaporative flux. The existing latent heat flux export will remain unchanged.

## Alternatives
If applicable, add a description of any alternative solutions or features you've considered.

## Testing:

1. Have you tested the code changes? On what platforms?
2. Have you run regression test in ufs-weather-model or ufs-s2s-model with code changes?
- Will the baseline results change?
- If yes, please give brief explanation of your code updates on the regression test.

Testing has been done in the coupled RTs. Baselines change because of round-off level differences for the evaporative flux in the first coupling loop. Full testing will be reported in the associated UFS PR.

## Dependent PRs:

Directly reference any issues or PRs in this or other repositories that this is related to, and describe how they are related. Example:
- required to support noaa-emc/GFDL_atmos_cubed_sphere/issues/<issue_number>
- ncar/ccpp-physics/pull/<pr_number>
- associated ufs-weather-model/pull/<pr_number>

### PR DESCRIPTION
**Pull request by DeniseWorthen**
## Description

Adds the evaporative flux to the available export fields from the ATM. 

Also adds a new procedure within the ``block_data_copy`` interface to prevent scaling of export values where the field contains a special value.

### Issue(s) addressed

- fixes #671 
- provides an incremental fix to resolution of CMEPS Issue https://github.com/NOAA-EMC/CMEPS/issues/93

## Testing

Testing results will be posted in the associated UFS PR https://github.com/ufs-community/ufs-weather-model/pull/1812

In brief, coupled tests using an active ATM and MOM6 will change baselines.

## Dependencies

If testing this branch requires non-default branches in other repositories, list them.
Those branches should have matching names (ideally)

- depends on https://github.com/NOAA-EMC/CMEPS/pull/96

Do PRs in upstream repositories need to be merged first?
If so add the "waiting for other repos" label and list the upstream PRs

- associated with https://github.com/ufs-community/ufs-weather-model/pull/1812

### CODE REVIEW

File: cpl/module_block_data.F90 [hunk 1]
```diff
@@ -138,25 +139,62 @@ subroutine block_copy_1d_r8_to_2d_r8(destin_ptr, source_ptr, block, block_index,
...
+    if (associated(destin_ptr) .and. associated(source_ptr)) then
+       !$omp parallel do private(ix,ib,jb,i,j)
+       do ix = 1, block%blksz(block_index)
+          ib = block%index(block_index)%ii(ix)
+          jb = block%index(block_index)%jj(ix)
+          i = ib - block%isc + 1
+          j = jb - block%jsc + 1
+          if (source_ptr(ix) .ne. special_value) then
+             destin_ptr(i,j) = scale_factor * source_ptr(ix)
+          else
```
Comment (junwang-noaa) [inline cpl/module_block_data.F90:172]: lines 171 and 172 are not needed?

Comment (DeniseWorthen) [reply cpl/module_block_data.F90:172]: I wasn't sure what value destin_ptr would have if it is not being filled with the source_ptr. I had originally 

```
j = jb - block%jsc + 1
destin_ptr(i,j) = special_value
if (source_ptr(ix) .ne. special_value) then
   destin_ptr(i,j) = .....
```

Comment (junwang-noaa) [reply cpl/module_block_data.F90:172]: I see.

**Review by DusanJovic-NOAA — APPROVED** [review]

**Review by junwang-noaa — APPROVED** [review]
Just to confirm the scaled fields are only used for coupling with ocn, not by aerosol models. Otherwise the code looks good to me.

**Comment by DeniseWorthen** [comment]
@junwang-noaa The scaling is only applied to the "mean" fields, which I don't believe are used anywhere else (for coupling itself). The full RTs have passed for all tests with only coupled tests changing. I will double check the first fields passed out of the ATM when aerosols are included.

**Comment by DeniseWorthen** [comment]
Fixing the scaling of the special value (huge) by rtime does change the fields coming out of the ATM at the first coupling timestep (dt=720s) when aerosols are included, but only at those points which contain the special value. They are otherwise identical. The four fields affected are the mean taux,y, sensible and latent. These fields have different maximum values (1.384612495609288E+34 vs 9.969209968386869E+36). 

The sfcf000.tileX.nc files are identical.

**Comment by junwang-noaa** [comment]
Do you still have those files that I can take a look? Thanks

**Comment by DeniseWorthen** [comment]
Yes, the develop output is in /scratch1/NCEPDEV/stmp2/Denise.Worthen/FV3_RT/rt_9995/s2swa.dev. This is a sandbox of the /scratch1/NCEPDEV/stmp2/Denise.Worthen/FV3_RT/rt_9995/cpld_control_p8_intel with only the executable changed.

**Comment by junwang-noaa** [comment]
I am curious, does the coupled test without aerosol have the same changes in the mean taux,y, sensible and latent?

**Comment by DeniseWorthen** [comment]
I've done another test using the current develop branch, but pointing to my FV3 branch. In this case, CMEPS is still using the laten_heat exported by the ATM so no answer changes are expected, but the exported fields will contain the impact of the scaling change in atmos_model. In my sandbox, s2swa.dev and s2swa.scale.dev have idential sfc and restart files. 

The mediator files (history and restart) show the only difference is the max value---the actual values are the same, they differ only in the areas which have a value huge. In one case it is the actual value huge, in the other it is huge*rtime.

**Comment by junwang-noaa** [comment]
Thanks for the testing and confirming the change is only on the max values.

**Pull request merged**